### ISSUE TITLE
Make validation tests more independent from specific library

### ISSUE THREAD

**Issue by Maarrk** [Shimul-Baidya/MBDyn#4]
I'd like to ask for two changes:
1. Don't check for the specific pydantic error, we just want to fail it in some way. I hope to add some validation logic directly in `_EntityBasePlaceholder` some day.

2. Skip if pydantic is unavailable, so when user runs the tests without this optional dependency we communicate "you can continue without it, but you'll miss some features"

Example of where I found it (not sure if it's the only place)
https://github.com/Shimul-Baidya/MBDyn/blob/80f359c36febf36195c5d767f7db161aa880a14e/contrib/PythonPreprocessor/test_MBDynLib.py#L166-L167

Example of how I wrote a similar test:
https://github.com/Shimul-Baidya/MBDyn/blob/80f359c36febf36195c5d767f7db161aa880a14e/contrib/PythonPreprocessor/test_MBDynLib.py#L51-L53

**Comment by Shimul-Baidya** [comment]
I've resolved the issue: [Relevant commit](https://github.com/Shimul-Baidya/MBDyn/commit/252cc33f99068fa046251f304ac58c99b95a4044)

Changes made:

1. Updated tests to avoid checking for specific pydantic errors, ensuring they fail generically instead.
2. Added logic to skip tests if pydantic is unavailable, allowing users to run tests without this optional dependency and still get meaningful feedback.
